### PR TITLE
manage demos using DemoCards.jl

### LINKED ISSUE #65: Introduce DemoCards.jl to demonstrations
I built a [`DemoCards.jl`](https://github.com/johnnychen94/DemoCards.jl) at the end of this summer as a tool to organize all demo files. This package is still at its early stage and a lot of features need to be implemented, but I'd like to open this issue as a notification in case some of you plan to do similar work.

A demo of `DemoCards.jl` can be found [here](https://johnnychen94.github.io/DemoCards.jl/dev/democards/gallery_of_packages/)

### PR DESCRIPTION
**Pull request by johnnychen94**
close #65

The core functionality of [`DemoCards.jl`](https://github.com/johnnychen94/DemoCards.jl) has been implemented. There might be some tiny bugs I missed, but I'd like to deploy it here to gather more feedback.

In summary, DemoCards support:

* folder structure is demo structure: we don't need to hard code demo structures in `make.jl`
* demos can be written in the format of markdown, and also _julia_: for julia demos, source file and notebooks(via nbviewer) are provided, and all results are generated on the fly (including cover image). This is partially powered by [`Literate.jl`](https://github.com/fredrikekre/Literate.jl).

Please check [here](https://johnnychen94.github.io/DemoCards.jl/stable/democards/quickstart/gallery/part2/simple_cameraman.html#simple-cameraman-1-1) for a live demo written in the Julia format.

Preview:

image
image
image

### CODE REVIEW

**Comment by johnnychen94** [comment]
The build fails because it requires [DemoCards v0.1.2](https://github.com/JuliaRegistries/General/pull/6118)

**Comment by johnnychen94** [comment]
I plan to merge this soon if there're no further comments. IMO in this way demos are way much easier to manage and write demos; assets are generated on the fly so we don't need to manually generate and update them.

File: Project.toml [hunk 1]
```diff
@@ -29,6 +30,7 @@ AxisArrays = "0"
 ColorTypes = "0"
 Colors = "0"
 CoordinateTransformations = "0"
+DemoCards = "0.1"
```
Comment (timholy) [inline Project.toml]:
```suggestion
DemoCards = "0.1.2"
```

**Comment by johnnychen94** [comment]
@timholy @juliohm @zygmuntszpak @Tokazama

I encourage you to take a try on this new demo system ([DemoCards.jl](https://github.com/johnnychen94/DemoCards.jl)) by contributing some demos and give some feedback. Since this package has a very poor coverage test, I plan to first put some of my efforts fixing potential bugs first and then start to contribute demos here.

**Comment by yakir12** [comment]
Wow, very very cool.

**Comment by timholy** [comment]
Thanks again for doing this, @johnnychen94!

It looks like several demo PRs came in while I was immersed in writing the debugger. We should migrate those and merge them.

I do wish other folks would be willing to merge such PRs. What is needed to get that to happen? If everything in JuliaImages depends on me, it won't be good for the long-term survival of the ecosystem. Do more people need merge privileges? Looks like there are only 3 of us now, I will add a few more.

**Comment by Tokazama** [comment]
>  If everything in JuliaImages depends on me, it won't be good for the long-term survival of the ecosystem.
I completely agree and having seen other imaging software in development I can say with a fair deal of confidence this all would have fallen apart in other languages without more regular contributors.

I'm trying to get to a stable point in all of my projects so I can come back and start contributing more. I'll be looking at how to get docs working for some of my personal projects and then maybe I can actually be helpful.

**Comment by johnnychen94** [comment]
This can be a JSoC 2020 project for others :D

**Pull request merged**